pcsx2/GS/Renderers/HW/GSRendererHW.cpp

```diff
@@ -863,6 +863,10 @@ bool GSRendererHW::IsSplitTextureShuffle()
 		return false;
 	}
 
+	// Different channel being shuffled, so needs to be handled separately (misdetection in 50 Cent)
+	if (m_vertex.buff[m_index.buff[0]].U != m_v.U)
+		return false;
+
 	// Check that both the position and texture coordinates are page aligned, so we can work in pages instead of coordinates.
 	// For texture shuffles, the U will be offset by 8.
 	const GSLocalMemory::psm_t& frame_psm = GSLocalMemory::m_psm[m_context->FRAME.PSM];
```

Comment:
This could've been moved a bit down, anyway it's fine, gonna shove it in a pr I'm working on.

Reply:
why bother moving it down? there's no need for it to be any lower.

Reply:
On the next return it's not used so we might set it for nothing.

Reply:
set what for nothing? I don't understand, if something is wrong it'll fail no matteer where you put it